pandas/core/indexes/multi.py

```diff
@@ -2321,65 +2320,33 @@ def reindex(self, target, method=None, level=None, limit=None, tolerance=None):
 
     def get_value(self, series, key):
         # Label-based
-        s = com.values_from_object(series)
-        k = com.values_from_object(key)
+        if not is_hashable(key) or is_iterator(key):
+            # We allow tuples if they are hashable, whereas other Index
+            #  subclasses require scalar.
+            # We have to explicitly exclude generators, as these are hashable.
+            raise InvalidIndexError(key)
 
         def _try_mi(k):
```

Comment:
should think about making try_mi non-nested (just for cleanliness), give it a better name and type in the future.

Reply:
yah there's plenty more to do here, but this is a big step